browser/ui/views/infobars/brave_confirm_infobar.cc

```diff
@@ -138,12 +142,26 @@ void BraveConfirmInfoBar::Layout(PassKey) {
     extra_button_->SizeToPreferredSize();
   }
 
+  CHECK(label_);
+  CHECK(link_);
+
   int x = GetStartX();
   Views views;
   views.push_back(label_.get());
   views.push_back(link_.get());
   AssignWidths(&views, std::max(0, GetEndX() - x - NonLabelWidth()));
 
+  if (GetDelegate()->HasMultiLineLabel()) {
+    const int label_width =
+        std::max(0, GetEndX() - x - NonLabelWidth() - link_->width());
+    label_->SizeToFit(label_width);
+
+    const int target_height = std::max(
+        label_->height(), ChromeLayoutProvider::Get()->GetDistanceMetric(
+                              DISTANCE_INFOBAR_HEIGHT));
+    SetTargetHeight(target_height);
+  }
+
   ChromeLayoutProvider* layout_provider = ChromeLayoutProvider::Get();
```

Comment:
nit - this can be moved to line 153 and `layout_provider` can be used in the above `if` clause.